services/retrieve_and_rank/v1.js

```diff
@@ -176,19 +182,232 @@ module.exports = function (RED) {
+                handleWatsonCallback(config.mode,node,msg,err,res);
+              });
+            } else {
+              var message = 'No cluster id specified';
+              node.error(message, msg)
+              return;
+            }           
+            break;
+          case 'delete':
+            if (params.cluster_id) {
+              retrieve_and_rank.deleteCluster(params,function(err,res) {
+                handleWatsonCallback(config.mode,node,msg,err,res);
+              });
+            } else {
+              var message = 'No cluster id specified';
+              node.error(message, msg)
+              return;
+            }
+            break;
+        }
+      });      
+    });
+  }
+
+  function uploadSolrConfigurationNode(config) {
+    RED.nodes.createNode(this, config);
+    var node = this;
+
+    this.on('input', function(msg) {
+      setupRankRetrieveNode(msg,config,this,function(retrieve_and_rank) {
+        var params = {
+          cluster_id: config.clusterid,
+          config_name: config.configname,
+          config_zip_path: config.configzippath
+        };
+        if (!params.cluster_id || !params.config_name || !params.config_zip_path) {
+          var message = 'No cluster id, configuration name or .zip file path specified';
+          node.error(message, msg)
+          return;
+        }
+        node.status({fill: 'blue', shape: 'ring', text: 'Uploading solr configuration...' });
+        retrieve_and_rank.uploadConfig(params, function (err, res) {
+          node.status({});
+          handleWatsonCallback(null,node,msg,err,res);
+        });
+      });      
+    });
+  }
+
+  function solrConfigurationSettingsNode(config) {
+    RED.nodes.createNode(this, config);
+    var node = this;
+
+    this.on('input', function(msg) {
+      setupRankRetrieveNode(msg,config,this,function(retrieve_and_rank) {
+        var params = {
+          cluster_id: config.clusterid,
+          config_name: config.configname
+        }
+        switch (config.mode) {
+          case 'list':
+            node.status({fill: 'blue', shape: 'ring', text: 'Getting solr configurations...' });
+            retrieve_and_rank.listConfigs(params,function(err,res) {
+              node.status({});
+              handleWatsonCallback(config.mode,node,msg,err,res);
+            });
+            break;
+          case 'info':
+            if (params.cluster_id && params.config_name) {
+              // retrieve_and_rank.getConfig(params, function(err,res) {
+              //   console.log("HELLO LADS GET CONFIG BACK");
+              //   console.log(err);
+              //   console.log(res);
+              // });
+              var configZipResponse = "hello world";
+              console.log("***********");
+              console.log(configZipResponse);
+              //Pass on response to payload as buffer
+              if (typeof(configZipResponse) == 'Buffer') {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Expected '===' and instead saw '=='.](https://www.codacy.com/app/rezgui-y/node-red-node-watson/file/2405715173/issues/source?bid=3225115&fileBranchId=3233651#l301)